release-22.2: backport time tables changes for DMS

otan opened the pull request:
Backport:
  * 1/1 commits from "sql: implement the `log_timezone` session variable" (#94123)
  * 1/1 commits from "sql: implement the pg_timezone_names table" (#94122)

Please see individual PRs for details.

/cc @cockroachdb/release

Release justification: improves DMS compat

otan committed:
sql: implement the `log_timezone` session variable

Release note (sql change): Add the `log_timezone` session variable,
which is read only and always UTC.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/94346)
<!-- Reviewable:end -->

otan committed:
sql: implement the pg_timezone_names table

Release note (sql change): Implement the `pg_timezone_names` pg_catalog
table, which lists all supported timezones.
